ini/INIReader.h

```diff
@@ -312,43 +324,38 @@ inline int INIReader::ParseError() const
 
 inline const std::set<std::string>& INIReader::Sections() const
 {
-    return _sections;
+    std::set<std::string> retval;
+    for (auto const& element : _values) {
+        retval.insert(element.first);
+    }
+    return retval;
+}
+
+inline const std::set<std::string>& INIReader::Keys(std::string section) const

```

Comment:
```
inline const std::set<std::string>& INIReader::Keys(std::string section) const
                                  ^
              this means return a reference.
```

In the above code, return value is a reference to the local variable `_setction `, which life time is limited in the function, so is invalid.

ref: https://stackoverflow.com/questions/4643713/c-returning-reference-to-local-variable